automation/run-tests.sh

```diff
@@ -68,17 +71,31 @@ function install_nmstate {
 }
 
 function run_tests {
-    docker_exec "
-      cd /workspace/nmstate &&
-      pytest \
-        --verbose --verbose \
-        --log-level=DEBUG \
-        --durations=5 \
-        --cov=\$($PYTHON_SITE_PATH_CMD)/libnmstate \
-        --cov=\$($PYTHON_SITE_PATH_CMD)/nmstatectl \
-        --cov-report=html:htmlcov-integ \
-        tests/integration \
-    ${nmstate_pytest_extra_args}"
+    if [ $TEST_TYPE == "all" ] || [ $TEST_TYPE == "tox_code_style" ];then
+        docker_exec 'cd /workspace/nmstate && tox -e flake8,pylint'
+    fi
+
+    if [ $TEST_TYPE == "all" ] || [ $TEST_TYPE == "tox_py27" ];then
+        docker_exec 'cd /workspace/nmstate && tox -e check-py27'
+    fi
+
+    if [ $TEST_TYPE == "all" ] || [ $TEST_TYPE == "tox_py36" ];then
+        docker_exec 'cd /workspace/nmstate && tox -e check-py36'
+    fi
+
+    if [ $TEST_TYPE == "all" ] || [ $TEST_TYPE == "integ" ];then
```

Comment:
The tox.ini also contains a definition to run integration tests for both python 2.7 and 3.6 - why are we defining how to run them differently here? Can we remove the definitions from tox or can we also update it and pass in the values for coverage in there somehow?

I just realized that for all, it will not execute all tests when an earlier one fails. IMHO it would be nice to run the integration tests first and ideally ensure that all tests run even with the earlier fail. If we can run the integration tests via tox, we could use just one tox call to run all tests, i.e. 'tox -e integration,flake8,pylint,check-py27, ...' to avoid early exits.

Reply:
That's a good point. Thanks. I will check whether we can fully depend on tox to kick off integration tests.

Reply:
When using tox, the nmstate is installed by pip. But we want to test the rpm, or should we both?

Reply:
[`skip_install=true`](https://tox.readthedocs.io/en/latest/config.html#conf-skip_install) should allow to avoid installing nmstate in the virtual env. For me it is enough to only test the RPM.